### DISCUSSION TITLE
React Wrapper

### DISCUSSION THREAD

**Discussion by LidiaKovac** [jaames/iro.js#258] [Show and tell]
Hi! I recently stumbled upon this AMAZING library while building my portfolio. I build a npm package as a react wrapper for it. It's my first time publishing an npm package that is supposed to work for more than just me, so I am very open to feedback and suggestions. 
Currently I am working on implementing other events other than `color:change` :) 

[Here](https://www.npmjs.com/package/react-iro) is the link to the package I have built. 
Thank you again for this amazing tool!

**Comment by jaames** [comment]
Hi @LidiaKovac! Thanks so much for tackling this!

I'm currently in the middle of rewriting iro.js, and as part of that I'm be moving over to using webcomponents which should hopefully make interoperability with various frameworks a lot more straightforward :)

This effort is still really appreciated, once the rewrite is done I think it would be great to have official iro.js wrapper packages and/or demos for frameworks like React, Vue, Svelte, Solid, etc. Please don't feel obligated to commit to anything but just out of curiosity, would you be interested in maintaining the React version of whatever that ends up being?

**Comment by LidiaKovac** [comment]
Hi! That would be cool! Absolutely keep me posted about it.

Il giorno mer 2 ago 2023 alle ore 00:30 James ***@***.***> ha
scritto:

> Hi Lidia! Thanks so much for tackling this!
>
> I'm currently in the middle of rewriting iro.js, and as part of that I'm
> be moving over to using webcomponents which should hopefully make
> interoperability with various frameworks a lot more straightforward :)
>
> This effort is still really appreciated, once the rewrite is done I think
> it would be great to have official iro.js wrapper packages and/or demos for
> frameworks like React, Vue, Svelte, Solid, etc. Please don't feel obligated
> to commit to anything but just out of curiosity, would you be interested in
> maintaining the React version of whatever that ends up being?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jaames/iro.js/discussions/258#discussioncomment-6610420>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQ3JY6WP6CYNX56I6CIQNTTXTF7QJANCNFSM6AAAAAA3ACGPJI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*Lidia Kovac || Digital Humanities Developer*
*LinkedIn <https://www.linkedin.com/in/lidia-kovac/> | GitHub
<https://github.com/LidiaKovac>*

**Reply by jaames** [reply]
@LidiaKovac Cool!! And sure thing! It'll take about two weeks for me to get the new version into a feature-complete state, then another month or two for me to bug-test and write up documentation etc.

I don't know if you'd prefer to wait until documentation is finished or not - a lot has changed! - I was planning to keep working on it in private until everything was stable but I'm happy to give you earlier access if you'd like to get a head start.

Not sure what's the best way of organising that, but I can be reached at @rkgkjr on discord, @jaames on telegram, or mail@jamesdaniel.dev

**Comment by LidiaKovac** [comment]
Hey :) Definetely give me a heads up when everything is over and I'll be
happy to make it a react component :D

Il giorno gio 10 ago 2023 alle ore 17:52 James ***@***.***>
ha scritto:

> @LidiaKovac <https://github.com/LidiaKovac> Cool!! And sure thing! It'll
> take about two weeks for me to get the new version into a feature-complete
> state, then another month or two for me to bug-test and write up
> documentation etc.
>
> I don't know if you'd prefer to wait until documentation is finished or
> not - a lot has changed! - I was planning to keep working on it in private
> until everything was stable but I'm happy to give you earlier access if
> you'd like to get a head start.
>
> Not sure what's the best way of organising that, but I can be reached at
> @rkgkjr on discord, @jaames <https://github.com/jaames> on telegram, or
> ***@***.***
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jaames/iro.js/discussions/258#discussioncomment-6693396>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQ3JY6RXWN4DSHBMJ25ZXFDXUT7U5ANCNFSM6AAAAAA3ACGPJI>
> .
> You are receiving this because you were mentioned.Message ID: <jaames/iro.
> ***@***.***>
>


-- 
*Lidia Kovac || Digital Humanities Developer*
*LinkedIn <https://www.linkedin.com/in/lidia-kovac/> | GitHub
<https://github.com/LidiaKovac>*
